gpm_api/tests/test_dataset/test_granule.py

```diff
@@ -297,59 +291,138 @@ def construct_dataset_and_check_validation(input_datetimes, expected_datetimes):
         construct_dataset_and_check_validation(datetimes_incomplete, datetimes)
 
 
-def test_finalize_dataset(monkeypatch):
+def _prepare_test_finalize_dataset(monkeypatch):
+    # Mock decoding coordinates
+    def mock_set_coordinates(ds, *args, **kwargs):
+        ds = ds.assign_coords({"decoding_coordinates": True})
+        return ds
+
+    monkeypatch.setattr(conventions, "set_coordinates", mock_set_coordinates)
+
+    # Return a default dataset
+    da = xr.DataArray(np.random.rand(1, 1), dims=("other", "along_track"))
```

Comment:
Maybe we should define here also a dataset for Grid data like IMERG that has `("time", "lat", "lon")` columns? 
And return the dataset according to the product?
Right now we test only for Orbit data ...

Reply:
The `_prepare_test_finalize_dataset` is not the only way I generate datasets for testing. Other dims are tested in `test_finalize_dataset_reshaping`, and it seemed to me that the rest of `finalize` did not depend on these dimensions. Tell me if you would still like to have the rest of `finalize` tested on Grid data

Reply:
Maybe it's worth still do this effort.